tests/sentry/ingest/billing_metrics_consumer/test_billing_metrics_consumer_kafka.py

```diff
@@ -1,28 +1,30 @@
 from datetime import datetime
 from unittest import mock
 
+import pytest
 from arroyo.backends.kafka import KafkaPayload
-from arroyo.types import Message, Partition, Topic
+from arroyo.processing.strategies.abstract import MessageRejected
+from arroyo.types import Message, Partition, Position, Topic
+from freezegun import freeze_time
 
-from sentry.constants import DataCategory
 from sentry.ingest.billing_metrics_consumer import (
     BillingTxCountMetricConsumerStrategy,
     MetricsBucket,
 )
 from sentry.sentry_metrics.indexer.strings import TRANSACTION_METRICS_NAMES
 from sentry.utils import json
-from sentry.utils.outcomes import Outcome
 
 
-@mock.patch("sentry.ingest.billing_metrics_consumer.track_outcome")
-def test_outcomes_consumed(track_outcome):
+@freeze_time("1985-10-26 21:00:00")
+@mock.patch(
+    "sentry.ingest.billing_metrics_consumer.BillingTxCountMetricConsumerStrategy._get_billing_producer"
+)
```

Comment:
Is this to avoid relying on Kafka which can be slow when testing? Arroyo ships with an in memory broker/producer/consumer implementation that is designed to allow for testing without Kafka which you might be interested in: https://github.com/getsentry/arroyo/tree/main/arroyo/backends/local

Reply:
I'm only interested in ensuring we call `produce` at the right time with the right arguments, and I assume the producer implementation arroyo provides is correct. Making an end-to-end test doesn't improve the coverage of the consumer itself, so I don't think using the in-memory stuff makes the test stronger?